Sources/PenguinParallel/NonblockingThreadPool/NonBlockingThreadPool.swift

```diff
@@ -384,6 +431,121 @@ extension NonBlockingThreadPool {
     }
   }
 
+  fileprivate func execute(_ task: PoolTask, localQueue: Queue) {
+    switch task {
+      case let .dispatch(op): op()
+      case let .join(task): executeJoinTask(task)
+      case let .parallelFor(slice): executeParallelForTaskSlice(slice, localQueue: localQueue)
+    }
+  }
+
+  fileprivate func executeJoinTask(_ task: UnsafeMutablePointer<JoinTask>) {
+    assert(!task.pointee.isDoneAcquiring(), "Join task done before even starting execution?!?")
```

Comment:
Do you want to use `assert` in benchmarking code, given that it will be compiled away in release mode?

Reply:
Hmm, not quite sure I follow? I do intentionally want to use an `assert` because I do want it compiled away in release mode. But having these asserts sprinkled throughout the code has been helpful in ensuring the implementation is correct during testing.

Reply:
That's fine!  I was just asking.